labelbox/client.py

```diff
@@ -512,14 +514,51 @@ def create_dataset(self, **kwargs):
         >>> dataset = client.create_dataset(name="<dataset_name>", projects=project)
 
         Args:
+            iam_integration (IAMIntegration) : Uses the default integration.
+                Optionally specify another integration or set as None to not use delegated access
             **kwargs: Keyword arguments with Dataset attribute values.
         Returns:
             A new Dataset object.
         Raises:
             InvalidAttributeError: If the Dataset type does not contain
                 any of the attribute names given in kwargs.
         """
-        return self._create(Dataset, kwargs)
+        dataset = self._create(Dataset, kwargs)
+
+        if iam_integration == IAMIntegration._DEFAULT:
+            iam_integration = self.get_organization(
+            ).get_default_iam_integration()
+
+        if iam_integration is None:
+            return dataset
+
+        if not isinstance(iam_integration, IAMIntegration):
+            raise TypeError(
+                f"iam integration must be a reference an `IAMIntegration` object. Found {type(iam_integration)}"
+            )
+
+        if not iam_integration.valid:
+            raise ValueError("Integration is not valid. Please select another.")
+        try:
+            self.execute(
+                """mutation setSignerForDatasetPyApi($signerId: ID!, $datasetId: ID!) {
+                    setSignerForDataset(data: { signerId: $signerId}, where: {id: $datasetId}){id}}
+                """, {
+                    'signerId': iam_integration.uid,
+                    'datasetId': dataset.uid
+                })
+            validation_result = self.execute(
+                """mutation validateDatasetPyApi($id: ID!){validateDataset(where: {id : $id}){
+                    valid checks{name, success}}}
+                """, {'id': dataset.uid})
+            if not validation_result['validateDataset']['checks'][0]['success']:
```

Comment:
checks[0] is the dataset validation and it always is run. Checks[1:] check that each data row is valid and since there are no data rows we only need to check checks[0]. But also looking into this more, validation_result['validateDataset']['valid'] returns true only if all checks are true. So I'll use that instead.